### PR TITLE
platform_manager: remove hardcoded platform checks for arp and dhcp monitors

### PR DESCRIPTION
**Pull request by tomereli**
Remove hard coded #ifdefs which doesn't start arp and dhcp monitors when
the BPL backend is Linux, where arp and dhcp monitors are not
implemented.
Instead, update the init_arp_monitor and init_dhcp_monitor functions to
return true if the BPL backend does not support monitoring.

While at it, also make dummy BWL available for non Linux platforms. This is useful in cases where initial bringup on a different OS (UGW, OpenWRT, etc.) should be done without WLAN bring up, and integration of the BWL can come later, like what will be done for the Nighthawk (RAX40) platform.

Signed-off-by: Tomer Eliyahu <tomer.b.eliyahu@intel.com>

### CODE REVIEW

File: framework/platform/bpl/include/bpl/bpl_err.h [hunk 1]
```diff
@@ -27,69 +27,71 @@ extern "C" {
...
+    ERROR_CODE(BPL_ERR_MONITOR_REPORT_PROCESS_FAIL)                                                \
+    ERROR_CODE(BPL_ERR_CONFIG_PLATFORM_REPORTED_INVALID_CONFIGURATION)                             \
+    ERROR_CODE(BPL_ERR_CONFIG_BACKHAUL_WIRED_INTERFACE_IS_UNSUPPORTED)                             \
+    ERROR_CODE(BPL_ERR_CONFIG_BACKHAUL_WIRELESS_INTERFACE_IS_UNSUPPORTED)                          \
+    ERROR_CODE(BPL_ERR_CONFIG_NO_VALID_BACKHAUL_INTERFACE)                                         \
+    ERROR_CODE(BPL_ERR_CONFIG_BRIDGE_INTERFACE_IS_EMPTY)                                           \
+    ERROR_CODE(BPL_ERR_WIFI_CONFIGURATION_CHANGE_TIMEOUT)                                          \
+    ERROR_CODE(BPL_ERR_WATCHDOG_PROCESS_STUCK)                                                     \
+    ERROR_CODE(BPL_ERR_WATCHDOG_PROCESS_ZOMBIE)                                                    \
+    ERROR_CODE(BPL_ERR_LAST)
```
Comment (morantr) [inline framework/platform/bpl/include/bpl/bpl_err.h:96]: As part of my changes on remove non-passive mode code, I deleted a lot of unused BPL error since it is not being used anymore, and aligned it as you did. Please add the new error codes at the end of the list so it would be easy to rebase.

File: framework/platform/bpl/linux/bpl_arp.cpp [hunk 2]
```diff
@@ -17,7 +17,7 @@
 
 using namespace beerocks::bpl;
 
-int bpl_arp_mon_start(BPL_ARP_MON_CTX *ctx, const char *iface) { return 0; }
+int bpl_arp_mon_start(BPL_ARP_MON_CTX *ctx, const char *iface) { return -2; }
```
Comment (arnout) [inline framework/platform/bpl/linux/bpl_arp.cpp:20]: Why don't we use `-BPL_ERR_OPERATION_NOT_SUPPORTED` here? (Not actually a change for this PR because we use int instead of enum everywhere, so fixing that would be a separate PR.)

File: framework/platform/bpl/include/bpl/bpl_err.h [hunk 3]
```diff
@@ -27,69 +27,71 @@ extern "C" {
...
-            BPL_ERR_CONFIG_PLATFORM_REPORTED_INVALID_CONFIGURATION) /* 57 */                       \
-        ERROR_CODE(BPL_ERR_CONFIG_BACKHAUL_WIRED_INTERFACE_IS_UNSUPPORTED) /* 58 */ ERROR_CODE(    \
-            BPL_ERR_CONFIG_BACKHAUL_WIRELESS_INTERFACE_IS_UNSUPPORTED) /* 59 */                    \
-        ERROR_CODE(BPL_ERR_CONFIG_NO_VALID_BACKHAUL_INTERFACE) /* 60 */ ERROR_CODE(                \
-            BPL_ERR_CONFIG_BRIDGE_INTERFACE_IS_EMPTY) /* 61 */                                     \
-        ERROR_CODE(BPL_ERR_WIFI_CONFIGURATION_CHANGE_TIMEOUT) /* 62 */ ERROR_CODE(                 \
-            BPL_ERR_WATCHDOG_PROCESS_STUCK) /* 63 */ ERROR_CODE(BPL_ERR_WATCHDOG_PROCESS_ZOMBIE)   \
-                                                                                                   \
-        /* 74 */ ERROR_CODE(BPL_ERR_LAST)
+    ERROR_CODE(BPL_ERR_NONE)                                                                       \
```
Comment (arnout) [inline framework/platform/bpl/include/bpl/bpl_err.h:30]: It took me a bit of figuring out what was going on here... Basically, the comment was removed, and this allowed clang-format to properly format it. An alternative would be to put the number as a second argument to the macro:

```
#define GENERATE_ERROR_ENUM(ENUM, VALUE) ENUM = VALUE,
#define GENERATE_ERROR_STRING(STRING, VALUE) #STRING "=" #VALUE,

#define FOREACH_ERROR_CODE(ERROR_CODE) \
    ERROR_CODE(BPL_ERR_NONE, 0) \
...
```

Anyway, this thing about the comment that is removed should have been part of the commit message.

(nitpick)

File: agent/src/beerocks/slave/platform_manager/platform_manager_thread.cpp [hunk 4]
```diff
@@ -2129,7 +2125,12 @@ bool main_thread::init_arp_monitor()
         m_uiArpMonIP   = info.ipa.s_addr;
         m_uiArpMonMask = info.nmask.s_addr;
 
-        if (bpl_arp_mon_start(&m_ctxArpMon, config.bridge_iface.c_str()) != 0) {
+        int ret = bpl_arp_mon_start(&m_ctxArpMon, config.bridge_iface.c_str());
+        if (ret < 0) {
+            if (ret == -BPL_ERR_OPERATION_NOT_SUPPORTED) {
+                LOG(INFO) << "Skip starting ARP monitor (not supported)";
+                return (true);
```
Comment (arnout) [inline agent/src/beerocks/slave/platform_manager/platform_manager_thread.cpp:2129]: I would add an explicit comment here to explain that it is not needed to initialize m_pArpRawSocket and m_pArpMonSocket and m_tpArpEntriesCleanup because they won't be used.

Also, maybe it would be better to not initialize m_uiArpMonIP and m_uiArpMonMask either in this case?

(nitpick)

File: framework/platform/bpl/include/bpl/bpl_err.h (hunk 1 above)
Comment (tomereli) [reply framework/platform/bpl/include/bpl/bpl_err.h:96]: The new error codes values have to be 1 and 2 so they can't be added to the end of the list. Rebase shouldn't be too hard even as it is now.

**Review by arnout — APPROVED** [review]
Just nitpicks

File: framework/platform/bpl/linux/bpl_arp.cpp (hunk 2 above)
Comment (tomereli) [reply framework/platform/bpl/linux/bpl_arp.cpp:20]: We can do this in a followup PR indeed.

File: framework/platform/bpl/include/bpl/bpl_err.h (hunk 3 above)
Comment (tomereli) [reply framework/platform/bpl/include/bpl/bpl_err.h:30]: Updated commit message accordingly. Didn't think it is worth the extra effort to manually add the enum value for each error code line, if we really need it we can do it in the future.

File: agent/src/beerocks/slave/platform_manager/platform_manager_thread.cpp (hunk 4 above)
Comment (tomereli) [reply agent/src/beerocks/slave/platform_manager/platform_manager_thread.cpp:2129]: I'm not sure I understand why this comment is needed. I do however agree about moving the m_uiArpMonIP and m_uiArpMonMask initialization - and did that.
